src/NServiceBus.TransactionalSession/TransactionalSession.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace NServiceBus.TransactionalSession
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Persistence;
+    using Transport;
+
+    class TransactionalSession : TransactionalSessionBase
+    {
+        public TransactionalSession(
+            ICompletableSynchronizedStorageSession synchronizedStorageSession,
+            IMessageSession messageSession,
+            IMessageDispatcher dispatcher) : base(synchronizedStorageSession, messageSession, dispatcher)
+        {
+        }
+
+        public override async Task Commit(CancellationToken cancellationToken = default)
+        {
+            await synchronizedStorageSession.CompleteAsync(cancellationToken).ConfigureAwait(false);
+
+            await dispatcher.Dispatch(new TransportOperations(pendingOperations.Operations), new TransportTransaction(), cancellationToken).ConfigureAwait(false);
```

Comment:
Also more of a nit. We can also do this later.

I was wondering if we could just have one `TransportTransaction` object that is created in the base class when needed and provided as a property.